### ISSUE TITLE
systemd-networkd-tests.py fails in 'test_bridge_configure_without_carrier'

### ISSUE THREAD

**Issue by mrc0mmand** [systemd/systemd#16105]
**systemd version the issue has been seen with**
> latest master

**Used distribution**
 > Arch Linux

I noticed that in several recent PRs the `systemd-networkd-tests.py` test fails with:

```

======================================================================
ERROR: test_bridge_configure_without_carrier (__main__.NetworkdBridgeTests) (test='no-slave')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./test/test-network/systemd-networkd-tests.py", line 2800, in test_bridge_configure_without_carrier
    self.wait_online(['bridge99:no-carrier'])
  File "./test/test-network/systemd-networkd-tests.py", line 506, in wait_online
    check_output(*args, env=env)
  File "./test/test-network/systemd-networkd-tests.py", line 49, in check_output
    return subprocess.check_output(command, universal_newlines=True, **kwargs).rstrip()
  File "/usr/lib/python3.8/subprocess.py", line 411, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  File "/usr/lib/python3.8/subprocess.py", line 512, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['/usr/lib/systemd/systemd-networkd-wait-online', '--timeout=20s', '--interface=bridge99:no-carrier']' returned non-zero exit status 1.

======================================================================
ERROR: test_bridge_configure_without_carrier (__main__.NetworkdBridgeTests) (test='add-slave')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./test/test-network/systemd-networkd-tests.py", line 2811, in test_bridge_configure_without_carrier
    self.wait_online(['bridge99:no-carrier:no-carrier'])
  File "./test/test-network/systemd-networkd-tests.py", line 506, in wait_online
    check_output(*args, env=env)
  File "./test/test-network/systemd-networkd-tests.py", line 49, in check_output
    return subprocess.check_output(command, universal_newlines=True, **kwargs).rstrip()
  File "/usr/lib/python3.8/subprocess.py", line 411, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  File "/usr/lib/python3.8/subprocess.py", line 512, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['/usr/lib/systemd/systemd-networkd-wait-online', '--timeout=20s', '--interface=bridge99:no-carrier:no-carrier']' returned non-zero exit status 1.

----------------------------------------------------------------------
Ran 138 tests in 2722.085s

FAILED (errors=2, expected failures=1)
```

Full log: https://ci.centos.org/job/systemd-pr-build-vagrant/7947/artifact//systemd-centos-ci/artifacts_kzRqbs/vagrant-logs.ZKt/vagrant-arch-testsuite.xm4/systemd-networkd-tests.py_FAIL.log
Machine journal: https://ci.centos.org/job/systemd-pr-build-vagrant/7947/artifact//systemd-centos-ci/artifacts_kzRqbs/vagrant-logs.ZKt/vagrant-journal-dump.log

(affecting, for example, #16096, #15729, ...).

/cc @yuwata, @ddstreet

**Comment by yuwata** [comment]
@mrc0mmand Hmm, if I remember correctly, we have not modified any code for that recently. have you update ArchLinux on the CI?

**Comment by mrc0mmand** [comment]
@yuwata the underlying image was definitely updated and I already fixed one of the issues it caused in #16081, but in this case I can't spot the issue.

**Comment by ddstreet** [comment]
Sigh, I *really* need to get the networkd tests merged so Ubuntu CI actually runs the systemd-networkd-tests.py tests :(

**Comment by mrc0mmand** [comment]
@yuwata unfortunately, the fix from #16108 doesn't seem to help in this case, since the test failed again few minutes ago: https://ci.centos.org/job/systemd-pr-build-vagrant-sanitizers/6232/artifact//systemd-centos-ci/artifacts_all/artifacts__d_iGZ/vagrant-logs.sZn/vagrant-arch-sanitizers-clang-testsuite.tDK/systemd-networkd_sanitizers_FAIL.log

**Comment by mrc0mmand** [comment]
Is there anything I could set in CentOS CI to make this more debuggable? Right now it's a quite common issue and I'm leaning towards disabling those tests, since it makes CI pretty unreliable.

**Comment by ddstreet** [comment]
let's see if #16165 lets this pass again, so we can take more time to figure out why it's not reaching 'configured' setup state without CI continuing to fail

**Comment by mrc0mmand** [comment]
Interestingly enough, the test keeps failing from time to time even after #16165 

```

======================================================================
ERROR: test_bridge_configure_without_carrier (__main__.NetworkdBridgeTests) (test='no-slave')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./test/test-network/systemd-networkd-tests.py", line 2801, in test_bridge_configure_without_carrier
    self.wait_online(['bridge99:no-carrier'], timeout='2m', setup_state=None)
  File "./test/test-network/systemd-networkd-tests.py", line 506, in wait_online
    check_output(*args, env=env)
  File "./test/test-network/systemd-networkd-tests.py", line 49, in check_output
    return subprocess.check_output(command, universal_newlines=True, **kwargs).rstrip()
  File "/usr/lib/python3.8/subprocess.py", line 411, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  File "/usr/lib/python3.8/subprocess.py", line 512, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['/usr/lib/systemd/systemd-networkd-wait-online', '--timeout=2m', '--interface=bridge99:no-carrier']' returned non-zero exit status 1.

======================================================================
ERROR: test_bridge_configure_without_carrier (__main__.NetworkdBridgeTests) (test='add-slave')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./test/test-network/systemd-networkd-tests.py", line 2812, in test_bridge_configure_without_carrier
    self.wait_online(['bridge99:no-carrier:no-carrier'], setup_state=None)
  File "./test/test-network/systemd-networkd-tests.py", line 506, in wait_online
    check_output(*args, env=env)
  File "./test/test-network/systemd-networkd-tests.py", line 49, in check_output
    return subprocess.check_output(command, universal_newlines=True, **kwargs).rstrip()
  File "/usr/lib/python3.8/subprocess.py", line 411, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  File "/usr/lib/python3.8/subprocess.py", line 512, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['/usr/lib/systemd/systemd-networkd-wait-online', '--timeout=20s', '--interface=bridge99:no-carrier:no-carrier']' returned non-zero exit status 1.

----------------------------------------------------------------------
Ran 138 tests in 2827.532s

FAILED (errors=2, expected failures=1)
```

[systemd-networkd-tests.py_FAIL.log](https://github.com/systemd/systemd/files/4824712/systemd-networkd-tests.py_FAIL.log)
[vagrant-journal-dump.log](https://github.com/systemd/systemd/files/4824713/vagrant-journal-dump.log)

**Comment by ddstreet** [comment]
> Interestingly enough, the test keeps failing from time to time even after #16165

well that's quite annoying :-/

I'll try to look at this asap to figure out what's going on.  I think it might be useful to update the systemd-networkd-tests.py to suppress some of the output for passing tests, as the failing test log data mostly gets buried.

**Comment by ddstreet** [comment]
I'm quite stumped at why it's failing; I opened another PR #16411 to try to avoid/workaround the failure (again), I guess let's see if that passes.

**Comment by yuwata** [comment]
I guess this is already fixed. Closing.
